pkg/cmd/update-check.go

```diff
@@ -17,31 +17,45 @@ package cmd
 import (
 	"encoding/json"
 	"fmt"
+	"io"
 	"io/ioutil"
+	"log"
 	"net/http"
+	"os"
+	"os/exec"
 
 	"github.com/Masterminds/semver"
+	"github.com/schollz/progressbar/v3"
 	"github.com/spf13/cobra"
 )
 
+const (
+	green = "\033[1;32m%s\033[0m"
+)
+
 // NewUpdateCheckCmd checks whether a newer version of gardenctl is available
 func NewUpdateCheckCmd() *cobra.Command {
 	return &cobra.Command{
 		Use:   "update-check",
 		Short: "Check whether new gardenctl version is available",
 		RunE: func(cmd *cobra.Command, args []string) error {
-
 			isAvailable, latestVersion, err := newVersionAvailable(version)
 			if err != nil {
 				return err
 			}
 			if isAvailable {
-				fmt.Printf("New version %q of Gardenctl is available at https://github.com/gardener/gardenctl/releases/latest \n", *latestVersion)
-				fmt.Println("Please get latest version from above URL and see https://github.com/gardener/gardenctl#installation for how to upgrade")
+				latestVersion := *latestVersion
+				fmt.Printf("New version %s of Gardenctl is available at https://github.com/gardener/gardenctl/releases/latest \n", latestVersion)
+				if askForConfirmation(green, "Do you want to install this version of Gardenctl (Y/N): ") {
+					out, err := exec.LookPath("gardenctl")
+					if err != nil {
+						log.Fatal(err)
+					}
+					download(out, "https://github.com/gardener/gardenctl/releases/download/"+latestVersion+"/gardenctl-darwin-amd64")
```

Comment:
Please, do not do this.
I am a linux user and this will not work for me.

Reply:
Thanks for the information. I will add extra logic for both user